manager/scheduler/scheduler.go

```diff
@@ -143,17 +143,21 @@ func (s *Scheduler) Run(ctx context.Context) error {
 		case event := <-updates:
 			switch v := event.(type) {
 			case api.EventCreateTask:
-				pendingChanges += s.createTask(ctx, v.Task)
+				s.createTask(ctx, v.Task)
+				tickRequired = true
 			case api.EventUpdateTask:
-				pendingChanges += s.updateTask(ctx, v.Task)
+				s.updateTask(ctx, v.Task)
+				tickRequired = true
```

Comment:
> `updateTask` returns nonzero if resources are being freed up

This is a little different from the original return value. It was the running task count from this update.